### ISSUE TITLE
Rewrite the back-end to be more stable and handle more hits

### ISSUE THREAD

**Issue by chriswmackey** [HydraShare/hydra#33]
I have noticed this issue for quite some time that the main hydrashare display page will include duplicate files at the bottom of the page and other example files that should be on this page do not display:
![image](https://cloud.githubusercontent.com/assets/5567574/24932520/de0b7ec4-1edf-11e7-9093-e12d9e3025d2.png)

For example, I cannot find my Urban Weather Generator file on the main page right now because of this.

I have noticed that the missing files change with time and I guess this main page is just failing to find all of the files that it should.  I just wanted to put this up here in case anyone had any ideas about it.  Maybe we should try refreshing this page every 30 minutes instead of 15 minutes to see if this has any effect.

**Comment by mostaphaRoudsari** [comment]
Hi @chriswmackey, These are two different issues! Where is the link to Urban Weather Generator file? In the back-end there is no duplication:

![image](https://cloud.githubusercontent.com/assets/2915573/24933941/049e7026-1ee7-11e7-9f40-dd1b71cd4912.png)

**Comment by mostaphaRoudsari** [comment]
Also here it is:
![image](https://cloud.githubusercontent.com/assets/2915573/24934101/a807a386-1ee7-11e7-9c50-43a8d3c59cc2.png)

**Comment by chriswmackey** [comment]
Right.  I said at the top of the issue that the number of duplicate files changes over time (as does the appearance/disappearance of other example files).  Every 15 minutes, we are erasing the record of the bug, which makes it a bit difficult to pin down.  It gets worst around mid-afternoon, I noticed.  I'll post again once I see it come back.
